loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts

```diff
@@ -150,11 +151,16 @@ class TransmissionHelper {
 
     private _addMesh(mesh: AbstractMesh): void {
         this._materialObservers[mesh.uniqueId] = mesh.onMaterialChangedObservable.add(this.onMeshMaterialChanged.bind(this));
-        if (this.shouldRenderAsTransmission(mesh.material)) {
-            this._transparentMeshesCache.push(mesh);
-        } else {
-            this._opaqueMeshesCache.push(mesh);
-        }
+        // we need to defer the processing because _addMesh may be called as part as an instance mesh creation, in which case some
+        // internal properties are not setup yet, like _sourceMesh (needed when doing mesh.material below)
+        Tools.SetImmediate(() => {
```

Comment:
Does this cause one frame to be rendered incorrectly?

Reply:
No, that causes a crash when doing `mesh.material` because `mesh._sourceMesh` is not setup yet (in the `InstancedMesh` constructor) when the `onNewMesh` observable is triggered.

Reply:
I think I didn't understand your question...

Yes, the first frame won't have the refraction texture setup for the transparent meshes. However, it is likely that the materials will also load some textures, so the material won't be ready the first few frame(s) anyway, meaning the meshes won't be rendered and so this problem won't show up.